nomenclature/region_mapping_models.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+from collections import Counter
+
+import yaml
+from jsonschema import validate
+from pydantic import BaseModel, validator, root_validator
+
+here = Path(__file__).parent.absolute()
+
+
+class NativeRegion(BaseModel):
+    name: str
+    rename: Optional[str]
+
+    @property
+    def target_native_region(self):
+        return self.rename if self.rename is not None else self.name
+
+
+class CommonRegion(BaseModel):
+    name: str
+    constituent_regions: List[NativeRegion]
+
+
+class RegionAggregationMapping(BaseModel):
+    model: str
+    native_regions: Optional[List[NativeRegion]]
+    common_regions: Optional[List[CommonRegion]]
+
+    @validator("native_regions")
+    def validate_native_regions(cls, v):
+        target_names = [nr.target_native_region for nr in v]
+        doubles = [item for item, count in Counter(target_names).items() if count > 1]
+        if doubles:
+            raise ValueError(
+                f"Two or more native regions share the same name: {doubles}"
+            )
```

Comment:
Minor suggestion for readability...
```suggestion
        duplicates = [item for item, count in Counter(target_names).items() if count > 1]
        if duplicates:
            raise ValueError(
                f"Two or more native regions share the same name: {duplicates}"
            )
```